app/controllers/discussions_controller.rb

```diff
@@ -22,6 +26,18 @@ def create
     end
   end
 
+  def update
+    @discussion = Discussion.find(params[:id])
+
+    respond_to do |format|
+      if can?(:update, @discussion) && @discussion.update(discussion_params)
+        format.html { redirect_to [@category, @discussion], notice: "Discussion was successfully updated." }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
```

Comment:
We should structure this like lock/unlock/pin/unpin where authorization returns a 403 instead of a 422,